dbt/adapters/bigquery/connections.py

```diff
@@ -713,8 +714,24 @@ def _query_and_results(
                 self._bq_job_link(query_job.location, query_job.project, query_job.job_id)
             )
 
-        iterator = query_job.result(max_results=limit, timeout=job_execution_timeout)
+        # only use async logic if user specifies a timeout
+        if job_execution_timeout:
+            loop = asyncio.new_event_loop()
+            future_iterator = asyncio.wait_for(
+                loop.run_in_executor(None, query_job.result), timeout=job_execution_timeout
```

Comment:
What happened to the `max_results=limit` kwarg? It's still in the non-`job_execution_timeout` route and I don't think the usage of it correlates with providing a timeout.